### PR TITLE
Persistence: Test Restructuring for Bitemporal Delta

### PR DESCRIPTION
**Pull request by prasar-ashutosh**
#### What type of PR is this?
Test Restructuring

#### What does this PR do / why is it needed ?
Test Restructuring for Bitemporal Delta

#### Which issue(s) this PR fixes:
1) Add validation while loading test data in H2
2) Define test scenarios and test cases for Bitemporal Delta
3) Add Bitemporal Delta tests in Memsql and Snowflake Modules

#### Does this PR introduce a user-facing change?
NO

### CODE REVIEW

File: legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-test/src/test/java/org/finos/legend/engine/persistence/components/BaseTest.java [hunk 1]
```diff
@@ -339,17 +343,68 @@
         .addFields(validityThroughTarget)
         .build();
 
+    protected SchemaDefinition bitemporalMainTableSchemaWithBatchIdAndTime = SchemaDefinition.builder()
+            .addFields(id)
+            .addFields(name)
+            .addFields(amount)
+            .addFields(validityFromReference)
+            .addFields(validityThroughReference)
```
Comment (kumuwu) [inline legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-test/src/test/java/org/finos/legend/engine/persistence/components/BaseTest.java:351]: Under our new setting, the main table should not have validityFromReference and validityThroughReference anymore

File: legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-test/src/test/java/org/finos/legend/engine/persistence/components/BaseTest.java [hunk 2]
```diff
@@ -339,17 +343,68 @@
...
+            .addFields(validityFromTarget)
+            .addFields(validityThroughTarget)
+            .build();
+
+    protected SchemaDefinition bitemporalMainTableSchemaWithDateTime = SchemaDefinition.builder()
+            .addFields(id)
+            .addFields(name)
+            .addFields(amount)
+            .addFields(validityFromReference)
+            .addFields(validityThroughReference)
```
Comment (kumuwu) [inline legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-test/src/test/java/org/finos/legend/engine/persistence/components/BaseTest.java:366]: Under our new setting, the main table should not have validityFromReference and validityThroughReference anymore

File: legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-test/src/test/java/org/finos/legend/engine/persistence/components/BaseTest.java (hunk 1 above)
Comment (prasar-ashutosh) [reply legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-test/src/test/java/org/finos/legend/engine/persistence/components/BaseTest.java]: Agreed, removed these two fields!

File: legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-test/src/test/java/org/finos/legend/engine/persistence/components/BaseTest.java (hunk 2 above)
Comment (prasar-ashutosh) [reply legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-test/src/test/java/org/finos/legend/engine/persistence/components/BaseTest.java]: Agreed, removed these two fields!

**Pull request closed without merging**